app/controllers/newflow/login_controller.rb

```diff
@@ -41,8 +41,24 @@ def logout
 
     protected ###############
 
+    def redirect_to_signup_if_go_param_present
+      if 'student_signup' == params[:go] && Settings::Db.store.student_feature_flag
```

Comment:
are all the variations in this boolean logic unit tested?

Reply:
No. But I believe it is covered in the feature tests. Refactored it though, looks much better to me:

```

    def redirect_to_signup_if_go_param_present
      if should_redirect_to_student_signup?
        redirect_to newflow_signup_student_path(request.query_parameters)
      elsif should_redirect_to_signup_welcome?
        redirect_to newflow_signup_path(request.query_parameters)
      end
    end

    def should_redirect_to_student_signup?
      params[:go] == GO_TO_STUDENT_SIGNUP && Settings::Db.store.student_feature_flag
    end

    def should_redirect_to_signup_welcome?
      params[:go] == GO_TO_SIGNUP && (Settings::Db.store.student_feature_flag || Settings::Db.store.educator_feature_flag)
    end
```